driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -431,7 +437,7 @@ CloseFuture removePool(Host host) {
         }
 
         // Wait pool creation before removing, so we don't lose connectivity
-        ListenableFuture<?> allPoolsCreatedFuture = Futures.successfulAsList(poolCreatedFutures);
+        ListenableFuture<?> allPoolsCreatedFuture = Futures.allAsList(poolCreatedFutures);
```

Comment:
Again this is not strictly related to JAVA-1431, but is required for errors to be propagated.